arkane/molpro.py

```diff
@@ -263,60 +263,85 @@ def loadConformer(self, symmetry=None, spinMultiplicity=0, opticalIsomers=1, sym
         """
-        f = open(self.path, 'r')
-        line=f.readline()
-        
-        # search for basisSet
-        while line!='':
-            if 'basis' in line.lower():
-                if 'vtz' in line.lower() or 'vdz' in line.lower():
-                    f12a=True
-                else: f12a=False
-                break
-            line=f.readline()
-        else: raise Exception('Could not find basis set in Molpro File')
-        # search for energy
-        E0=None
-        if f12a:
-            while line!='':
-                if ('RHF-UCCSD(T)-F12a energy' in line
-                        or 'RHF-RCCSD(T)-F12a energy' in line
-                        or 'CCSD(T)-F12a total energy  ' in line):
-                    E0=float(line.split()[-1])
-                    break
-                if 'Electronic Energy at 0' in line:
-                    E0=float(line.split()[-2])
-                    break
-                line=f.readline()
-        else:
-            while line!='':
-                if ('RHF-UCCSD(T)-F12b energy' in line
-                        or 'RHF-RCCSD(T)-F12b energy' in line
-                        or 'CCSD(T)-F12b total energy  ' in line):
-                    E0=float(line.split()[-1])
-                    break
-                if 'Electronic Energy at 0' in line:
-                    E0=float(line.split()[-2])
+        E0 = None
+        with open(self.path, 'r') as f:
+            lines = f.readlines()
+            # Determine whether this is b12a or f12b according to the basis set, or whether this is MRCI.
+            f12a, f12b, mrci = False, False, False
+            for line in lines:
+                if 'basis' in line.lower():
+                    if 'vtz' in line.lower() or 'vdz' in line.lower():
+                        f12a = True  # MRCI could also have a vdz/vtz basis, so don't break yet
+                    elif any(high_basis in line.lower() for high_basis in ['vqz', 'v5z', 'v6z', 'v7z', 'v8z']):
+                        f12b = True  # MRCI could also have a v(4+)z basis, so don't break yet
+                elif 'mrci' in line.lower():
+                    mrci = True
+                    f12a, f12b = False, False
                     break
-                line=f.readline()
-
-        f.close()
-        logging.debug('Molpro energy found is {0} hartree'.format(E0))
+                elif 'point group' in line.lower():
+                    # We should know the method by this point, so break if possible, but don't throw an error yet
+                    if any([mrci, f12a, f12b]):
+                        break
+            else:
+                raise ValueError('Could not determine type of calculation. Currently, CCSD(T)-F12a, CCSD(T)-F12b,'
+                                 ' MRCI, MRCI+Davidson are supported')
+            # Search for E0
+            for line in lines:
+                if f12a:
+                    if ('RHF-UCCSD(T)-F12a energy' in line
```

Comment:
Can these all be grouped into a single check? I.e., just this:
```suggestion
                    if 'CCSD(T)-F12a energy' in line:
```

Reply:
I don't know. Are there cases where we'll have `CCSD(T)-F12a total energy  ` and no `CCSD(T)-F12a energy`? I can probably group the first two checks, perhaps we should leave the third?

Reply:
OK. Perhaps we should also check F12 with a VQZ basis: If not specifying F12b in the input file, but just F12, will it also output something like `RHF-UCCSD(T)-F12a energy`? If so, we should differentiate them when parsing

Reply:
Yeah, I still meant that we should differentiate. I.e., have two tests like we currently do, one that does `if CCSD(T)-F12a in line and energy in line` and one that does `if CCSD(T)-F12b in line and energy in line`.